### ISSUE TITLE
[FEA] Switch default `chunksize` for `dask_cudf.read_csv` to `2gb` from `256 MiB`

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#6560]
**Is your feature request related to a problem? Please describe.**
I think we should switch `chunksize` for `dask_cudf.read_csv` increase our `chunksize` from `256 MiB` to something bigger like `1 gb` to `2gb` . 

The code change will go at line : 

https://github.com/rapidsai/cudf/blob/c382989330f97715e4aec3f2c677b0a6d2e99647/python/dask_cudf/dask_cudf/io/csv.py#L14

## Why:

From experience on `tpcx-bb`, workflows and other users,  it seems like current default of  `256 MiB`: 

a. Is often too low to saturate GPU meaningfully especially with wider frames for both io and downstream operations. 

b.  Bigger chunk size often just sidesteps various communication and scheduler bottlenecks for shuffle operations like repartition/merge etc.

CC: @beckernick , @ayushdg , @quasiben , @kkraus14 .

**Comment by kkraus14** [comment]
One note here is that per thread default stream / explicit CUDA streams could likely change this where we can run multiple threads on a GPU and saturate it more effectively via concurrent work instead of relying on only a single task to saturate it.

This is of course at the cost of a larger number of tasks and more work for the scheduler to handle the larger number of tasks.

**Comment by VibhuJawa** [comment]
> One note here is that per thread default stream / explicit CUDA streams could likely change this where we can run multiple threads on a GPU and saturate it more effectively via concurrent work instead of relying on only a single task to saturate it.
> 
> This is of course at the cost of a larger number of tasks and more work for the scheduler to handle the larger number of tasks.

Agreed that we will have to revisit this default when we have explicit CUDA streams but for now, I feel we should be safe to increase the default chunk size.

If we all end up agreeing on changing this default, happy to do a quick pr to fix this.

**Comment by kkraus14** [comment]
The problem with straight up changing it to 2GB is if someone is using say a 4GB GPU they're going to run into problems. Can we somehow make it something like: `min(2GB, (1/8)*gpu_mem)`?

**Comment by VibhuJawa** [comment]
> The problem with straight up changing it to 2GB is if someone is using say a 4GB GPU they're going to run into problems. Can we somehow make it something like: `min(2GB, (1/8)*gpu_mem)`?

As this happens in the client process, I am unsure of a clean way to query `gpu_mem`,  as the worker processes might have different memory than the client.  Maybe @quasiben  will know. 

If we can make the simplifying assumption here that `gpu_mem` is the memory of the client process this should be doable.

**Comment by harrism** [comment]
Also, it's available / free GPU mem, not total memory, right? That also depends on what RMM memory resource you are using, and not all are queryable...

**Comment by kkraus14** [comment]
It would be based on available, not free. Dask manages spilling as needed so you don't want to change configuration based on the amount of GPU memory available generally.

**Comment by harrism** [comment]
In my book, "available" and "free" memory are the same thing.

**Comment by kkraus14** [comment]
> In my book, "available" and "free" memory are the same thing.

Sorry, by "available" I interpreted as the configured maximum pool size (if applicable) or the total amount of GPU memory.

**Comment by rjzamora** [comment]
NVTabular uses pynvml to get the total memory, and uses that to set `chunksize`.  We used to use rmm (understandably) dropped reliable support for this.

**Comment by beckernick** [comment]
@VibhuJawa is this still relevant?

**Comment by VibhuJawa** [comment]
> @VibhuJawa is this still relevant?

Given the above problems i think changing default is not an option in the short term. 

That said may be we need a best-practices page like dask has (see [doc]( https://docs.dask.org/en/latest/dataframe-best-practices.html )) to inform users about the best practices around chunk size .

**Comment by vyasr** [comment]
I'm going to close this as stale for now. We can reopen in the context of newer engines/GPUs/our code if we think it's useful.
